[Bug][azure devops] got error when restart

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.


### What happened

time="2023-05-15 14:04:10" level=error msg="failed to load plugins
        caused by: get error when invoking remote function run-migrations
        Wraps: (2) remote error response:
          | Traceback (most recent call last):
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
          |     self.dialect.do_execute(
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
          |     cursor.execute(statement, parameters)
          | psycopg2.errors.SyntaxError: syntax error at or near "PRIMARY"
          | LINE 1: ALTER TABLE _tool_azuredevops_jobs DROP PRIMARY KEY
          |                                                 ^
          |
          |
          | The above exception was the direct cause of the following exception:
          |
          | Traceback (most recent call last):
          |   File "/app/python/plugins/azuredevops/azuredevops/main.py", line 132, in <module>
          |     AzureDevOpsPlugin.start()
          |   File "/app/python/pydevlake/pydevlake/plugin.py", line 265, in start
          |     fire.Fire(PluginCommands(plugin))
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/fire/core.py", line 141, in Fire
          |     component_trace = _Fire(component, args, parsed_flag_args, context, name)
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/fire/core.py", line 466, in _Fire
          |     component, remaining_args = _CallAndUpdateTrace(
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
          |     component = fn(*varargs, **kwargs)
          |   File "/app/python/pydevlake/pydevlake/ipc.py", line 54, in wrapper
          |     ret = func(self, *args)
          |   File "/app/python/pydevlake/pydevlake/ipc.py", line 105, in run_migrations
          |     self._plugin.run_migrations(create_db_engine(db_url), force)
          |   File "/app/python/pydevlake/pydevlake/plugin.py", line 120, in run_migrations
          |     model.migrate(session)
          |   File "/app/python/plugins/azuredevops/azuredevops/models.py", line 129, in migrate
          |     session.execute(f'ALTER TABLE {self.__tablename__} DROP PRIMARY KEY')
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlmodel/orm/session.py", line 101, in execute
          |     return super().execute(  # type: ignore
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1712, in execute
          |     result = conn._execute_20(statement, params or {}, execution_options)
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1705, in _execute_20
          |     return meth(self, args_10style, kwargs_10style, execution_options)
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 333, in _execute_on_connection
          |     return connection._execute_clauseelement(
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
          |     ret = self._execute_context(
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
          |     self._handle_dbapi_exception(
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2124, in _handle_dbapi_exception
          |     util.raise_(
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 208, in raise_
          |     raise exception
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
          |     self.dialect.do_execute(
          |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
          |     cursor.execute(statement, parameters)
          | sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "PRIMARY"
          | LINE 1: ALTER TABLE _tool_azuredevops_jobs DROP PRIMARY KEY
          |                                                 ^
          |
          | [SQL: ALTER TABLE _tool_azuredevops_jobs DROP PRIMARY KEY]
          | (Background on this error at: https://sqlalche.me/e/14/f405)
        Wraps: (3) exit status 1
        Wraps: (4) exit status 1
        Error types: (1) *hintdetail.withDetail (2) *hintdetail.withDetail (3) *hintdetail.withDetail (4) *exec.ExitError"
panic: get error when invoking remote function run-migrations
Wraps: (2) remote error response:
  | Traceback (most recent call last):
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
  |     self.dialect.do_execute(
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
  |     cursor.execute(statement, parameters)
  | psycopg2.errors.SyntaxError: syntax error at or near "PRIMARY"
  | LINE 1: ALTER TABLE _tool_azuredevops_jobs DROP PRIMARY KEY
  |                                                 ^
  |
  |
  | The above exception was the direct cause of the following exception:
  |
  | Traceback (most recent call last):
  |   File "/app/python/plugins/azuredevops/azuredevops/main.py", line 132, in <module>
  |     AzureDevOpsPlugin.start()
  |   File "/app/python/pydevlake/pydevlake/plugin.py", line 265, in start
  |     fire.Fire(PluginCommands(plugin))
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/fire/core.py", line 141, in Fire
  |     component_trace = _Fire(component, args, parsed_flag_args, context, name)
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/fire/core.py", line 466, in _Fire
  |     component, remaining_args = _CallAndUpdateTrace(
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/fire/core.py", line 681, in _CallAndUpdateTrace
  |     component = fn(*varargs, **kwargs)
  |   File "/app/python/pydevlake/pydevlake/ipc.py", line 54, in wrapper
  |     ret = func(self, *args)
  |   File "/app/python/pydevlake/pydevlake/ipc.py", line 105, in run_migrations
  |     self._plugin.run_migrations(create_db_engine(db_url), force)
  |   File "/app/python/pydevlake/pydevlake/plugin.py", line 120, in run_migrations
  |     model.migrate(session)
  |   File "/app/python/plugins/azuredevops/azuredevops/models.py", line 129, in migrate
  |     session.execute(f'ALTER TABLE {self.__tablename__} DROP PRIMARY KEY')
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlmodel/orm/session.py", line 101, in execute
  |     return super().execute(  # type: ignore
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1712, in execute
  |     result = conn._execute_20(statement, params or {}, execution_options)
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1705, in _execute_20
  |     return meth(self, args_10style, kwargs_10style, execution_options)
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 333, in _execute_on_connection
  |     return connection._execute_clauseelement(
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
  |     ret = self._execute_context(
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
  |     self._handle_dbapi_exception(
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2124, in _handle_dbapi_exception
  |     util.raise_(
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 208, in raise_
  |     raise exception
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
  |     self.dialect.do_execute(
  |   File "/root/.cache/pypoetry/virtualenvs/azuredevops-ANnMAkq9-py3.9/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
  |     cursor.execute(statement, parameters)
  | sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "PRIMARY"
  | LINE 1: ALTER TABLE _tool_azuredevops_jobs DROP PRIMARY KEY
  |                                                 ^
  |
  | [SQL: ALTER TABLE _tool_azuredevops_jobs DROP PRIMARY KEY]
  | (Background on this error at: https://sqlalche.me/e/14/f405)
Wraps: (3) exit status 1
Wraps: (4) exit status 1
Error types: (1) *hintdetail.withDetail (2) *hintdetail.withDetail (3) *hintdetail.withDetail (4) *exec.ExitError

goroutine 1 [running]:
github.com/apache/incubator-devlake/server/services.Init()
        /app/server/services/init.go:97 +0x229
github.com/apache/incubator-devlake/server/api.CreateApiService()
        /app/server/api/api.go:63 +0x3d
main.main()
        /app/server/main.go:44 +0x8a

### What do you expect to happen

I don't know what happend. I think maybe there a bug in python migration.

### How to reproduce

1. deploy devlake by docker-compose.yml
2. stop and re-deploy.

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/backend/python/plugins/azuredevops/azuredevops/models.py
+++ b/backend/python/plugins/azuredevops/azuredevops/models.py
@@ -126,5 +126,11 @@ class Job(ToolModel, table=True):
 
     @classmethod
     def migrate(self, session: Session):
-        session.execute(f'ALTER TABLE {self.__tablename__} DROP PRIMARY KEY')
+        dialect = session.bind.dialect.name
+        if dialect == 'mysql':
+            session.execute(f'ALTER TABLE {self.__tablename__} DROP PRIMARY KEY')
+        elif dialect == 'postgresql':
+            session.execute(f'ALTER TABLE {self.__tablename__} DROP CONSTRAINT {self.__tablename__}_pkey')
+        else:
+            raise Exception(f'Unsupported dialect {dialect}')
         session.execute(f'ALTER TABLE {self.__tablename__} ADD PRIMARY KEY (id, build_id)')

